server/src/domain/move/move.service.ts

```diff
@@ -0,0 +1,27 @@
+import { MoveEntity } from '@app/infra/entities/move.entity';
+import { Inject, Injectable } from '@nestjs/common';
+import { IMoveRepository } from './move.repository';
+
+@Injectable()
+export class MoveService {
+  constructor(@Inject(IMoveRepository) private repository: IMoveRepository) {}
+
+  async handleCleanup() {
+    for (const move of await this.repository.getDeletedMoves()) {
+      if (this.isReadyForDeletion(move)) {
+        this.repository.delete(move);
+      }
+    }
+
+    return true;
```

Comment:
Is this needed? It won't ever return anything but true, which is a bit redundant.

Reply:
JobHandler must map to `boolean | Promise<boolean>`, so yeah